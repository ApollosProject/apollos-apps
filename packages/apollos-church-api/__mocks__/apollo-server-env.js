import fetch from 'jest-fetch-mock';

import { Constants } from 'apollos-church-api/src/connectors/rock';
import * as apolloDatasourceMocks from './apollo-datasource-mocks';
import * as rockMocks from './rock-api-mocks';

const apolloServerEnv = require.requireActual('apollo-server-env');

// eslint-disable-next-line
const Response = apolloServerEnv.Response;

const resolveWith = (data, url) =>
  Promise.resolve(
    new Response(JSON.stringify(data), {
      url,
      status: 200,
      statusText: 'OK',
      headers: new apolloServerEnv.Headers({
        'Content-Type': 'application/json',
      }),
    })
  );

fetch.mockLiveDataSourceApis = () => {
  fetch.mockImplementation((request) => {
    if (request.url.match('/api/v1/events/current')) {
      return resolveWith(apolloDatasourceMocks.liveStreamLive(), request.url);
    }
    if (request.url.match('/passage/html')) {
      return resolveWith(apolloDatasourceMocks.ESVScripture(), request.url);
    }
    return Promise.reject();
  });
};

fetch.mockRockDataSourceAPI = () => {
  fetch.mockImplementation((request) => {
    let { url } = request;
    url = decodeURI(url);
    if (!url.match(Constants.ROCK_API)) return Promise.reject();

    if (url.match('api/ContentChannels/\\d')) {
      return resolveWith(rockMocks.contentChannel());
    }

    if (url.match('api/ContentChannels')) {
      return resolveWith([rockMocks.contentChannel()]);
    }

    if (url.match('api/ContentChannelItems/test-case-no-attributes')) {
      const mock = rockMocks.contentItem();
      mock.AttributeValues = {};
      mock.Attributes = {};
      return resolveWith(mock);
    }

    if (url.match('api/ContentChannelItems/\\d')) {
      return resolveWith(rockMocks.contentItem());
    }

    if (url.match('api/ContentChannelItems')) {
      return resolveWith([rockMocks.contentItem()]);
    }

    if (url.match('api/ContentChannelItemAssociations')) {
      return resolveWith([rockMocks.contentChannelItemAssociation()]);
    }

    if (url.match('api/Auth/Login')) {
      const body = JSON.parse(request.body);
      const response = new Response('');
      if (body.Password === 'good') {
        response.headers.set('set-cookie', 'some cookie');
        return Promise.resolve(response);
      }
      return Promise.reject({ ...response, status: 401 });
    }

    if (url.match('api/People/GetCurrentPerson')) {
      return resolveWith(rockMocks.people());
    }

    if (url.match('api/People/\\d')) {
      return resolveWith(rockMocks.people());
    }

    if (url.match('api/People')) {
      if (request.method === 'POST') {
        const { Email } = JSON.parse(request.body);
        if (!Email) {
          const response = new Response('');
          response.status = 400;
          return Promise.reject(response);
        }
        return resolveWith({ personId: 35 });
      }

      return resolveWith([rockMocks.people()]);
    }

    if (url.match('api/UserLogins')) {
      if (request.method === 'POST') {
        const { UserName } = JSON.parse(request.body);
        if (!UserName) {
          const response = new Response('');
          response.status = 400;
          return Promise.reject(response);
        }
        return resolveWith({ id: 21 });
      }
      const identity = url // identity = UserName
        .split('eq')
        .pop()
        .trim(' '); // EXAMPLE URL: /api/UserLogins?$filter=UserName eq 'isaac.hardy@newspring.cc'

      if (!identity) {
        const response = new Response('');
        response.status = 400;
        return Promise.reject(response);
      }
      if (identity === `'isaac.hardy@newspring.cc'`)
        return resolveWith([rockMocks.userLogins()]);

      return resolveWith([]);
    }

    return Promise.reject();
  });
};

const apolloServerEnvMocked = {
  ...apolloServerEnv,
  Response: fetch.Response,
  fetch,
};

module.exports = apolloServerEnvMocked;
