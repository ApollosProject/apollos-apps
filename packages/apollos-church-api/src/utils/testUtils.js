import getContext from 'apollos-church-api/src/getContext';
import getDataSources from 'apollos-church-api/src/getDataSources';
import { KeyValueCache } from 'apollo-server-caching';

export function getTestContext(req) {
  const context = getContext(req);

  const dataSources = getDataSources();
  // Apollo Server does this internally.
  Object.values(dataSources).forEach((dataSource) => {
    if (dataSource.initialize) {
      dataSource.initialize({ context, cache: KeyValueCache });
    }
  });
  context.dataSources = dataSources;
  return context;
}

export const buildGetMock = (response, dataSource) => {
  const get = jest.fn();
  if (Array.isArray(response) && Array.isArray(response[0])) {
    response.forEach((responseVal) => {
      get.mockReturnValueOnce(
        new Promise((resolve) => resolve(dataSource.normalize(responseVal)))
      );
    });
  }
  get.mockReturnValue(
    new Promise((resolve) => resolve(dataSource.normalize(response)))
  );
  return get;
};
