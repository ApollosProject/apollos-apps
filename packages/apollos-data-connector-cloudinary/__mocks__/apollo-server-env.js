import fetch from 'jest-fetch-mock';

import * as apolloDatasourceMocks from './apollo-datasource-mocks';

const apolloServerEnv = require.requireActual('apollo-server-env');

// eslint-disable-next-line
const Response = apolloServerEnv.Response;

const resolveWith = (data, url) =>
  Promise.resolve(
    new Response(JSON.stringify(data), {
      url,
      status: 200,
      statusText: 'OK',
      headers: new apolloServerEnv.Headers({
        'Content-Type': 'application/json',
      }),
    })
  );

fetch.mockLiveDataSourceApis = () => {
  fetch.mockImplementation((request) => {
    if (request.url.match('api.scripture.api.bible')) {
      return resolveWith(apolloDatasourceMocks.Scripture(), request.url);
    }
    return Promise.reject();
  });
};

const apolloServerEnvMocked = {
  ...apolloServerEnv,
  Response: fetch.Response,
  fetch,
};

module.exports = apolloServerEnvMocked;
