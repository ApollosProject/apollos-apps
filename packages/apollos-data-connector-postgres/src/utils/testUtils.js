/* eslint-disable import/prefer-default-export */
import createMigrationRunner from '../postgres/performMigrations';
import { sequelize } from '../postgres';

export const setupPostgresTestEnv = async (models = [], context) => {
  if (context.church.slug !== 'global') {
    await setupPostgresTestEnv(models, { church: { slug: 'global' } });
  }

  try {
    // If we don't have a set of migrations.
    // Hacky way of saying - have the migrations been run?
    await sequelize
      .getQueryInterface()
      .sequelize.query('select * from "SequelizeMeta"');
  } catch (e) {
    const migrationRunner = await createMigrationRunner({
      logger: null,
    });
    await migrationRunner.up();
  }

  await Promise.all(
    models.map((m) => m.models.createModel && m.models.createModel(context))
  );

  await Promise.all(
    models.map((m) => m.models.setupModel && m.models.setupModel(context))
  );
};
