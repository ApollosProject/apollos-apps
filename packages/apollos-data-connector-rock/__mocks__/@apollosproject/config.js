const ApolloServer = require.requireActual('@apollosproject/config').default;

ApolloServer.loadJs({
  ROCK_MAPPINGS: {
    CONTENT_ITEM: {
      ContentSeriesContentItem: {
        EntityType: 'ContentChannelItem',
      },
      DevotionalContentItem: {
        EntityType: 'SomeOtherEntity',
      },
      MediaContentItem: {
        EntityType: 'ContentChannelItem',
      },
      UniversalContentItem: {
        EntityType: 'ContentChannelItem',
      },
      ContentItem: {
        EntityType: 'ContentChannelItem',
      },
    },
  },
  ROCK: {
    TIMEZONE: 'America/New_York',
    API_URL: 'https://apollosrock.newspring.cc/api',
    API_TOKEN: 'some-rock-token',
    IMAGE_URL: 'https://apollosrock.newspring.cc/GetImage.ashx',
  },
});

export default ApolloServer;
