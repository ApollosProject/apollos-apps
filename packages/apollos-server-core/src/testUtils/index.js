/* eslint-disable import/prefer-default-export */
import { makeExecutableSchema } from 'apollo-server';
import { testSchema } from '@apollosproject/data-schema';
import { KeyValueCache } from 'apollo-server-caching';
import { createApolloServerConfig } from '..';

export const createTestHelpers = (models) => {
  const serverConfig = createApolloServerConfig(models);
  const getContext = (req) => {
    const testContext = serverConfig.context(req);
    const testDataSources = serverConfig.dataSources();
    // Apollo Server does this internally.
    Object.values(testDataSources).forEach((dataSource) => {
      if (dataSource.initialize) {
        dataSource.initialize({ context: testContext, cache: KeyValueCache });
      }
    });
    testContext.dataSources = testDataSources;
    return testContext;
  };
  const getSchema = (schemas = []) =>
    makeExecutableSchema({
      typeDefs: [...serverConfig.schema, ...schemas, testSchema],
      resolvers: serverConfig.resolvers,
      resolverValidationOptions: {
        requireResolversForResolveType: false,
      },
    });

  return { getSchema, getContext };
};
