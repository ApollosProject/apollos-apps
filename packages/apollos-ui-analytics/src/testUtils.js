import React from 'react';

import { Providers } from '@apollosproject/ui-kit';
import { ApolloProvider } from 'react-apollo';
import { ApolloClient } from 'apollo-client';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { createHttpLink } from 'apollo-link-http';
import AnalyticsProvider from './Provider';

const cache = new InMemoryCache();

export const trackMock = jest.fn();

const client = new ApolloClient({
  link: createHttpLink({ fetch: jest.fn() }),
  cache,
  resolvers: {},
});

export const wait = (time) =>
  new Promise((resolve) => setTimeout(resolve, time));

// eslint-disable-next-line
export const ProviderStack = ({ children }) => (
  <ApolloProvider client={client}>
    <Providers>
      <AnalyticsProvider trackFunctions={[trackMock]}>
        {children}
      </AnalyticsProvider>
    </Providers>
  </ApolloProvider>
);
