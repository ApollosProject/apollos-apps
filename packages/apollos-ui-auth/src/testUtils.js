/* eslint-disable import/prefer-default-export */
import React from 'react';
import { Providers as UIProviders } from '@apollosproject/ui-kit';
import { MockedProvider } from 'react-apollo/test-utils';

// eslint-disable-next-line
export const Providers = ({ children, ...props }) => (
  <UIProviders {...props}>
    <MockedProvider {...props}>{children}</MockedProvider>
  </UIProviders>
);
