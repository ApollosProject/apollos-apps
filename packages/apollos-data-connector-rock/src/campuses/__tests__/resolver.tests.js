import { graphql } from 'graphql';
import { fetch } from 'apollo-server-env';
import ApollosConfig from '@apollosproject/config';
import { createGlobalId } from '@apollosproject/server-core';
import { createTestHelpers } from '@apollosproject/server-core/lib/testUtils';
import { campusSchema, authSchema } from '@apollosproject/data-schema';

// we import the root-level schema and resolver so we test the entire integration:
import * as Campus from '../index';
import { Person } from '../../index';

class AuthDataSourceMock {
  initialize = () => {};

  getCurrentPerson = () => ({
    id: 51,
    firstName: 'Isaac',
    lastName: 'Hardy',
    nickName: 'Isaac',
    email: 'isaac.hardy@newspring.cc',
    photo: {
      url:
        'https://apollosrock.newspring.cc:443/GetImage.ashx?guid=60fd5f35-3167-4c26-9a30-d44937287b87',
    },
  });

  getCurrentPersonAlternateLookupId = () => '0faad2f-3258f47';
}

const Auth = {
  schema: authSchema,
  dataSource: AuthDataSourceMock,
  resolver: { Query: { currentUser: () => ({ profile: { id: 51 } }) } },
};

const { getContext, getSchema } = createTestHelpers({ Campus, Person, Auth });

ApollosConfig.loadJs({
  ROCK: {
    API_URL: 'https://apollosrock.newspring.cc/api',
    API_TOKEN: 'some-rock-token',
    IMAGE_URL: 'https://apollosrock.newspring.cc/GetImage.ashx',
  },
});

describe('Campus', () => {
  let schema;
  let context;
  beforeEach(() => {
    fetch.resetMocks();
    fetch.mockRockDataSourceAPI();
    schema = getSchema([campusSchema]);
    context = getContext();
  });

  it('gets a campus by id', async () => {
    const query = `
      query {
        node(id: "${createGlobalId(1, 'Campus')}") {
          ...on Campus {
            id
            name
            street1
            street2
            city
            state
            postalCode
            latitude
            longitude
            image {
              uri
            }
            distanceFromLocation
          }
        }
      }
    `;
    const rootValue = {};

    const result = await graphql(schema, query, rootValue, context);
    expect(result).toMatchSnapshot();
  });

  it('gets a campus by id with location', async () => {
    const query = `
      query {
        node(id: "${createGlobalId(1, 'Campus')}") {
          ...on Campus {
            id
            name
            street1
            street2
            city
            state
            postalCode
            latitude
            longitude
            image {
              uri
            }
            distanceFromLocation(location: { latitude: 34, longitude: -82 })
          }
        }
      }
    `;
    const rootValue = {};

    const result = await graphql(schema, query, rootValue, context);
    expect(result).toMatchSnapshot();
  });

  it('gets campuses sorted by location', async () => {
    const query = `
      query {
        campuses(location: { latitude: 34.59814, longitude: -82.62045 }) {
            id
            name
            street1
            street2
            city
            state
            postalCode
            latitude
            longitude
            image {
              uri
            }
            distanceFromLocation
        }
      }
    `;
    const rootValue = {};

    const result = await graphql(schema, query, rootValue, context);
    expect(result).toMatchSnapshot();
  });

  it('gets campuses (without location)', async () => {
    const query = `
      query {
        campuses {
            id
            name
            street1
            street2
            city
            state
            postalCode
            latitude
            longitude
            image {
              uri
            }
            distanceFromLocation
        }
      }
    `;
    const rootValue = {};

    const result = await graphql(schema, query, rootValue, context);
    expect(result).toMatchSnapshot();
  });

  it("gets current user's campus", async () => {
    const query = `
      query {
        currentUser {
          profile {
            campus {
              id
              name
              latitude
              longitude
            }
          }
        }
      }
    `;
    const rootValue = {};

    const getMock = jest.fn(() =>
      Promise.resolve([
        {
          campus: {
            id: 1,
            name: 'the best campus',
            location: { latitude: 1.1, longitude: 2.2 },
          },
        },
      ])
    );

    context.dataSources.Campus.get = getMock;

    const result = await graphql(schema, query, rootValue, context);
    expect(result).toMatchSnapshot();
    expect(getMock.mock.calls).toMatchSnapshot();
  });

  it('returns null if a user has no campus', async () => {
    const query = `
      query {
        currentUser {
          profile {
            campus {
              id
              name
            }
          }
        }
      }
    `;
    const rootValue = {};

    const getMock = jest.fn(() => Promise.resolve([]));

    context.dataSources.Campus.get = getMock;

    const result = await graphql(schema, query, rootValue, context);
    expect(result).toMatchSnapshot();
    expect(getMock.mock.calls).toMatchSnapshot();
  });

  it("updates a current user's campus", async () => {
    const query = `
      mutation {
        updateUserCampus(campusId: "${createGlobalId(123, 'Campus')}") {
          campus {
            id
            name
          }
        }
      }
    `;
    const rootValue = {};

    const getMock = jest.fn(() =>
      Promise.resolve([
        { id: 1, campus: { id: 123, name: 'the very best campus' } },
      ])
    );

    const patchMock = jest.fn(() => Promise.resolve());

    context.dataSources.Campus.get = getMock;
    context.dataSources.Campus.patch = patchMock;

    const result = await graphql(schema, query, rootValue, context);
    expect(result).toMatchSnapshot();
    expect(getMock.mock.calls).toMatchSnapshot();
    expect(patchMock.mock.calls).toMatchSnapshot();
  });
});
