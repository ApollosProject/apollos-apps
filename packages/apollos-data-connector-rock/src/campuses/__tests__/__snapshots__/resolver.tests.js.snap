// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Campus gets a campus by id 1`] = `
Object {
  "data": Object {
    "node": Object {
      "city": "Anderson",
      "distanceFromLocation": null,
      "id": "Campus:559b23fd0aa90e81b1c023e72e230fa1",
      "image": Object {
        "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=a47e8e69-9542-42de-9ed3-ff42fdc82417",
      },
      "latitude": 34.59814,
      "longitude": -82.62045,
      "name": "Anderson Campus",
      "postalCode": "29621-3619",
      "state": "SC",
      "street1": "2940 Concord Rd",
      "street2": "",
    },
  },
}
`;

exports[`Campus gets a campus by id with location 1`] = `
Object {
  "data": Object {
    "node": Object {
      "city": "Anderson",
      "distanceFromLocation": 54.42261893027277,
      "id": "Campus:559b23fd0aa90e81b1c023e72e230fa1",
      "image": Object {
        "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=a47e8e69-9542-42de-9ed3-ff42fdc82417",
      },
      "latitude": 34.59814,
      "longitude": -82.62045,
      "name": "Anderson Campus",
      "postalCode": "29621-3619",
      "state": "SC",
      "street1": "2940 Concord Rd",
      "street2": "",
    },
  },
}
`;

exports[`Campus gets campuses (without location) 1`] = `
Object {
  "data": Object {
    "campuses": Array [
      Object {
        "city": "Anderson",
        "distanceFromLocation": null,
        "id": "Campus:559b23fd0aa90e81b1c023e72e230fa1",
        "image": Object {
          "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=a47e8e69-9542-42de-9ed3-ff42fdc82417",
        },
        "latitude": 34.59814,
        "longitude": -82.62045,
        "name": "Anderson Campus",
        "postalCode": "29621-3619",
        "state": "SC",
        "street1": "2940 Concord Rd",
        "street2": "",
      },
      Object {
        "city": "Richardson",
        "distanceFromLocation": null,
        "id": "Campus:965b6e6d7046a885bea4e300b5c0400d",
        "image": Object {
          "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=5f9fd97a-fd6e-4ae5-ab5d-a9b95290d9b6",
        },
        "latitude": 32.95103,
        "longitude": -96.74738,
        "name": "Dallas Campus",
        "postalCode": "75080-5525",
        "state": "TX",
        "street1": "102 N Weatherred Dr",
        "street2": "",
      },
      Object {
        "city": "Cincinnati",
        "distanceFromLocation": null,
        "id": "Campus:4f68015ba18662a7409d1219a4ce013e",
        "image": Object {
          "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=bad9eb54-3e78-44a7-a028-3f2d6e71a36f",
        },
        "latitude": 39.10501,
        "longitude": -84.51138,
        "name": "Cincinnati Campus",
        "postalCode": "45202-2118",
        "state": "OH",
        "street1": "120 E 8th St",
        "street2": "",
      },
      Object {
        "city": "South Barrington",
        "distanceFromLocation": null,
        "id": "Campus:a0f64573eabf00a607bec911794d50fb",
        "image": Object {
          "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=79a0e51b-8c6b-4d54-9601-cb87383b7be8",
        },
        "latitude": 42.09203,
        "longitude": -88.13289,
        "name": "Chicago Campus",
        "postalCode": "60010-6143",
        "state": "IL",
        "street1": "67 Algonquin Rd",
        "street2": "",
      },
    ],
  },
}
`;

exports[`Campus gets campuses sorted by location 1`] = `
Object {
  "data": Object {
    "campuses": Array [
      Object {
        "city": "Anderson",
        "distanceFromLocation": 0,
        "id": "Campus:559b23fd0aa90e81b1c023e72e230fa1",
        "image": Object {
          "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=a47e8e69-9542-42de-9ed3-ff42fdc82417",
        },
        "latitude": 34.59814,
        "longitude": -82.62045,
        "name": "Anderson Campus",
        "postalCode": "29621-3619",
        "state": "SC",
        "street1": "2940 Concord Rd",
        "street2": "",
      },
      Object {
        "city": "Cincinnati",
        "distanceFromLocation": 328.43647347848844,
        "id": "Campus:4f68015ba18662a7409d1219a4ce013e",
        "image": Object {
          "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=bad9eb54-3e78-44a7-a028-3f2d6e71a36f",
        },
        "latitude": 39.10501,
        "longitude": -84.51138,
        "name": "Cincinnati Campus",
        "postalCode": "45202-2118",
        "state": "OH",
        "street1": "120 E 8th St",
        "street2": "",
      },
      Object {
        "city": "South Barrington",
        "distanceFromLocation": 597.4083397282817,
        "id": "Campus:a0f64573eabf00a607bec911794d50fb",
        "image": Object {
          "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=79a0e51b-8c6b-4d54-9601-cb87383b7be8",
        },
        "latitude": 42.09203,
        "longitude": -88.13289,
        "name": "Chicago Campus",
        "postalCode": "60010-6143",
        "state": "IL",
        "street1": "67 Algonquin Rd",
        "street2": "",
      },
      Object {
        "city": "Richardson",
        "distanceFromLocation": 818.552029472811,
        "id": "Campus:965b6e6d7046a885bea4e300b5c0400d",
        "image": Object {
          "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=5f9fd97a-fd6e-4ae5-ab5d-a9b95290d9b6",
        },
        "latitude": 32.95103,
        "longitude": -96.74738,
        "name": "Dallas Campus",
        "postalCode": "75080-5525",
        "state": "TX",
        "street1": "102 N Weatherred Dr",
        "street2": "",
      },
    ],
  },
}
`;

exports[`Campus gets current user's campus 1`] = `
Object {
  "data": Object {
    "currentUser": Object {
      "profile": Object {
        "campus": Object {
          "id": "Campus:559b23fd0aa90e81b1c023e72e230fa1",
          "latitude": 1.1,
          "longitude": 2.2,
          "name": "the best campus",
        },
      },
    },
  },
}
`;

exports[`Campus gets current user's campus 2`] = `
Array [
  Array [
    "/Groups/GetFamilies/51?loadAttributes=expanded&%24expand=Campus%2CCampus%2FLocation%2CCampus%2FLocation%2FImage&%24top=1",
    Object {},
    Object {},
  ],
]
`;

exports[`Campus returns null if a user has no campus 1`] = `
Object {
  "data": Object {
    "currentUser": Object {
      "profile": Object {
        "campus": null,
      },
    },
  },
}
`;

exports[`Campus returns null if a user has no campus 2`] = `
Array [
  Array [
    "/Groups/GetFamilies/51?loadAttributes=expanded&%24expand=Campus%2CCampus%2FLocation%2CCampus%2FLocation%2FImage&%24top=1",
    Object {},
    Object {},
  ],
]
`;

exports[`Campus updates a current user's campus 1`] = `
Object {
  "data": Object {
    "updateUserCampus": Object {
      "campus": Object {
        "id": "Campus:2cc8f539e345e8f479a55e2fdc9743c0",
        "name": "the very best campus",
      },
    },
  },
}
`;

exports[`Campus updates a current user's campus 2`] = `
Array [
  Array [
    "/Groups/GetFamilies/51?loadAttributes=expanded&%24top=1",
    Object {},
    Object {},
  ],
  Array [
    "/Groups/GetFamilies/51?loadAttributes=expanded&%24expand=Campus%2CCampus%2FLocation%2CCampus%2FLocation%2FImage&%24top=1",
    Object {},
    Object {},
  ],
]
`;

exports[`Campus updates a current user's campus 3`] = `
Array [
  Array [
    "/Groups/1",
    Object {
      "CampusId": "123",
    },
  ],
]
`;
